rocAL/rocAL_pybind/examples/rocAL_api_coco_pipeline.py

```diff
@@ -62,106 +60,64 @@ def __init__(self, pipelines, tensor_layout=types.NCHW, reverse_channels=False,
         self.device_id = self.loader._device_id
         self.bs = self.loader._batch_size
         self.num_anchors = num_anchors
-        self.w = self.loader.getOutputWidth()
-        self.h = self.loader.getOutputHeight()
-        self.n = self.loader.getOutputImageCount()
-        self.rim = self.loader.getRemainingImages()
-        self.display = display
-        print("____________REMAINING IMAGES____________:", self.rim)
-        color_format = self.loader.getOutputColorFormat()
-        self.p = (1 if color_format is types.GRAY else 3)
-        if tensor_layout == types.NCHW:
-            if self.device == "cpu":
-                if self.tensor_dtype == types.FLOAT:
-                    self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.float32)
-                elif self.tensor_dtype == types.FLOAT16:
-                    self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.float16)
-            else:
-                torch_gpu_device = torch.device('cuda', self.device_id)
-                if self.tensor_dtype == types.FLOAT:
-                    self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.float32, device=torch_gpu_device)
-                elif self.tensor_dtype == types.FLOAT16:
-                    self.out = torch.empty((self.bs*self.n, self.p, int(self.h/self.bs), self.w,), dtype=torch.float16, device=torch_gpu_device)
-        else:
-            if self.device == "cpu":
-                if self.tensor_dtype == types.FLOAT:
-                    self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.float32)
-                elif self.tensor_dtype == types.FLOAT16:
-                    self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.float16)
-            else:
-                torch_gpu_device = torch.device('cuda', self.device_id)
-                if self.tensor_dtype == types.FLOAT:
-                    self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.float32, device=torch_gpu_device)
-                elif self.tensor_dtype == types.FLOAT16:
-                    self.out = torch.empty((self.bs*self.n, int(self.h/self.bs), self.w, self.p), dtype=torch.float16, device=torch_gpu_device)
-
-        #Image id of a batch of images
-        self.image_id = np.zeros(self.bs, dtype="int32")
-        # Count of labels/ bboxes in a batch
-        self.bboxes_label_count = np.zeros(self.bs, dtype="int32")
+        #Number of batch size handled by each GPU
+        self.len = math.ceil(self.loader.getRemainingImages()/self.bs)
         # Image sizes of a batch
-        self.img_size = np.zeros((self.bs * 2), dtype="int32")
+        self.img_size = np.zeros((self.bs * 3), dtype="int32")
```

Comment:
Why are we multiplying by a scalar value 3?

Reply:
I overlooked this Shobana, Not sure guess it was present in here already. Changed it back to 2 , as we width & height